dali/python/nvidia/dali/plugin/jax.py

```diff
@@ -140,13 +143,15 @@ def __init__(
             auto_reset=False,
             last_batch_padded=False,
             last_batch_policy=LastBatchPolicy.FILL,
-            prepare_first_batch=True):
+            prepare_first_batch=True,
+            sharding=None):
```

Comment:
```suggestion
            sharding: jax.sharding.Sharding=None):
```
Small suggestion, you may consider adding type hint here

Reply:
`jax.sharding.Sharding` is abstract base class. This should accept `NamesSharding`, `PositionalSharding` and maybe others in the future.

When it comes to inheritance and type hints this should be done as 
```python
from typing import Type
...

sharding; Type[jax.sharding.Sharding]=None):
```

I added straight up type assertion against  `NamesSharding` and `PositionalSharding` since these are the ones we are testing against. If there is new type of sharding in the future we will add it and add tests for it. 

We are not using type hints anywhere so I didn't want to add them just here.